ivy/functional/backends/paddle/statistical.py

```diff
@@ -212,7 +240,30 @@ def cumprod(
     dtype: Optional[paddle.dtype] = None,
     out: Optional[paddle.Tensor] = None,
 ) -> paddle.Tensor:
-    raise IvyNotImplementedException()
+    dtype = ivy.as_native_dtype(dtype)
```

Comment:
the dtype handling part will not yield the expected results, you are casting to the closeset valid dtype which will return the same dtype because all of them are valid, also there's no need for infer dtype,`x.dtype` is enough and will result in native dtype, try to implement it using ivy.swapaxes